### PR TITLE
Fix `Filename.quote_command` hanlding of forward slashes in program path under Win32

### LINKED ISSUE #13415: `Filename.quote_command` on Win32 does not handle forward slashes in program path correctly
Forward slashes in the "program" argument of `Filename.quote_command` should be taken as a reason to quote the program path. Otherwise, `cmd.exe` interprets forward slashes as introducing a flag (eg `foo/x` is interpreted as `foo /x`). The issue is exhibited by the following interaction:
```
$ rlwrap ocaml
OCaml version 4.14.2+LEXIFI
Enter #help;; for help.

# print_endline (Filename.quote_command "a/b.exe" ["x"]);;
"a/b.exe ^"x^""
- : unit = ()
# Sys.command (Filename.quote_command "a/b.exe" ["x"]);;
'a' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.
- : int = 1
#
```
Here `a/b.exe` should be quoted (surrounded by double quotes) but it isn't, so `cmd.exe` think the program name path is `a` instead of `a/b.exe`.

### PR DESCRIPTION
**Pull request by nojb**
Fixes #13415 

Tweak the code to quote the program path if it contains a forward slash.

### CODE REVIEW

**Comment by nojb** [comment]
(The CI failure is unrelated.)

**Comment by dra27** [comment]
For cmd, I think we still have some fairly tight limits on command length to worry about - might it be better to convert forward slashes to backslashes instead?

**Comment by nojb** [comment]
> For cmd, I think we still have some fairly tight limits on command length to worry about - might it be better to convert forward slashes to backslashes instead?

Yes. Also, this quoting solution may not work in all cases, see the discussion at https://github.com/ocaml/dune/pull/10869. I will revise the patch.

**Comment by nojb** [comment]
> For cmd, I think we still have some fairly tight limits on command length to worry about - might it be better to convert forward slashes to backslashes instead?

I did this. Do you mind taking a look? Thanks!

File: testsuite/tests/lib-filename/quotecommand.ml [hunk 1]
```diff
@@ -100,3 +101,9 @@ let _ =
                "in voluptate"; "-out"; "velit esse cillum"; "-err"; "dolore"];
   cat_file "my file.tmp"; Sys.remove "my file.tmp";
   Sys.remove "my echo.exe"
+
+let _ =
+  if not (Sys.file_exists "sub") then Sys.mkdir "sub" 0o777;
+  copy_file "myecho.exe" "sub/myecho.exe";
+  printf "-------- Forward slashes in program position\n";
+  run "sub/myecho.exe" ["alea iacta est"]
```
Comment (damiendoligez) [inline testsuite/tests/lib-filename/quotecommand.ml]: You could simplify like this:
```suggestion
  printf "-------- Forward slashes in program position\n";
  run "./myecho.exe" ["alea iacta est"]
```

**Review by damiendoligez — APPROVED** [review]
Looks good. I have a small suggestion to simplify the test.

File: testsuite/tests/lib-filename/quotecommand.ml (hunk 1 above)
Comment (nojb) [reply testsuite/tests/lib-filename/quotecommand.ml]: Suggestion applied. Thanks @damiendoligez!

**Comment by nojb** [comment]
Cherry-picked to 5.3: https://github.com/ocaml/ocaml/commit/8a1ba112569399779336fc555145630253c0cdf3

**Pull request merged**